### PR TITLE
Lazily initialize dexag client

### PR DESCRIPTION
**Pull request by e00E**
We want to avoid doing heavy work on construction.

### CODE REVIEW

File: driver/src/price_estimation/dexag/mod.rs [hunk 1]
```diff
@@ -57,13 +81,26 @@ where
     Api: DexagApi + Sync,
 {
     fn get_prices(&self, tokens: &[Token]) -> Result<HashMap<TokenId, u128>> {
+        if tokens.is_empty() {
+            return Ok(HashMap::new());
+        }
+
+        self.initialize_if_needed()
```
Comment (fleupold) [inline driver/src/price_estimation/dexag/mod.rs]: Can this method return the api token list as part of the result? This way we don't have to borrow and unwrap.

**Review by fleupold — COMMENTED** [review]
Thanks for tackling this! What about using https://docs.rs/lazy-init/0.3.0/lazy_init/struct.Lazy.html? Looks like an abstract enough issue to be generalised.

File: driver/src/price_estimation/dexag/mod.rs (hunk 1 above)
Comment (e00E) [reply driver/src/price_estimation/dexag/mod.rs]: Agree that it's nicer. Implemented.

**Comment by e00E** [comment]
I did not know about `lazy_init`. After trying it out, it is does not fit in this case because our initialization can fail whereas the crate expects it to never fail.

**Comment by fleupold** [comment]
> whereas the crate expects it to never fail

I saw this too, but was thinking we can just make it return a Result<HashMap>

**Review by fleupold — APPROVED** [review]
No strong feeling about using the library, up to you...

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2F3o7Z4tNIWOQWTlfGN2%2Fgiphy.gif&token=green_light" data-gitmeme-token="green_light"></a>

**Comment by e00E** [comment]
But then if it returns an error the first time I won't update it the second time because it thinks that is already the correct result.

**Pull request merged**